tests/conftest.py

```diff
@@ -1,42 +1,83 @@
 # SPDX-License-Identifier: MIT
 
+import os
+import os.path
+import shutil
+import stat
+import tempfile
+
 import pytest
 
 
-DUMMY_PYPROJECT = '''
-[build-system]
-requires = ['flit_core']
-build-backend = 'flit_core.buildapi'
-'''.strip()
+def pytest_addoption(parser):
+    parser.addoption('--run-integration', action='store_true', help='run the integration tests')
+    parser.addoption('--only-integration', action='store_true', help='only run the integration tests')
+
+
+def pytest_collection_modifyitems(config, items):
+    skip_int = pytest.mark.skip(reason='integration tests not run')
+    skip_other = pytest.mark.skip(reason='only integration tests are run')
+
+    for item in items:
+        if os.path.basename(item.location[0]) == 'test_integration.py':
+            if not config.getoption('--run-integration') and not config.getoption('--only-integration'):
+                item.add_marker(skip_int)
+        elif config.getoption('--only-integration'):
+            item.add_marker(skip_other)
+
+
+@pytest.fixture
+def packages_path():
+    return os.path.realpath(os.path.join(__file__, '..', 'packages'))
```

Comment:
there's some duplication here, maybe define a ``HERE`` global viarable that contains this folder.

Reply:
We can do that. I usually avoid doing small optimizations like that when it's like two instances. If you notice here, that de-duplication would actually result in a bigger file. It also adds a complexity factor of having to check, or remember, what `HERE` is when we read the code.

Btw, I am not saying this just to be picky or something. I am establishing a precedent for these cases, this is also what I am doing for most other comments. This way we can come up to a conclusion and be on the same page for similar things in the future.

Reply:
> or remember, what `HERE` is when we read the code.

It really would be just 3-5 lines above, so not that hard to find out; and in all projects I've been seing HERE used as the folder of the file, so it's kinda self-documenting. So I see a lot of precedence on going down this path.

Reply:
Yeah, I understand, but that is not the main point I am arguing against. That being that we only have two instances and with the code needed to declare the variable, the file would actually be bigger :sweat_smile:. The complexity point is just something I always consider in these situations, it might be pretty small, but it's there. I have also seen simple things like this accumulate to a point where it does make a difference.

With that said, if you really want, I can change it.